### PR TITLE
CSI nodeSelector, affinity

### LINKED ISSUE #842: CSI: NodeSelector
**Is your feature request related to a problem? Please describe.**
I would like the chance to specify a nodeSelector for the CSI daemonset too.
It exists for the injector deployment and the vault server but for the CSI deamonset

**Additional context**
Look at the `templates/injector-deployment.yaml`  and `templates/server-statefulset.yaml`, they includes the nodeSelector.
I'd like the same for `templates/csi-daemonset.yaml`

### PR DESCRIPTION
**Pull request by Toninh0**
This helps configuring the scheduling of CSI daemonset pods

Closes #842

### CODE REVIEW

**Comment by aurelien-wefight** [comment]
Hello ! :) Any update for this PR ?

File: values.yaml [hunk 1]
```diff
@@ -1057,6 +1057,19 @@ csi:
...
+    # nodeSelector labels for csi pod assignment, formatted as a multi-line string or YAML map.
+    # ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
+    # Example:
+    # nodeSelector:
+    #   beta.kubernetes.io/arch: amd64
+    nodeSelector: []
+
+    # Affinity Settings
+    # Commenting out or setting as empty the affinity variable, will allow
+    # deployment to single node services such as Minikube
```
Comment (tomhjp) [inline values.yaml:1069]: Doesn't seem relevant, as the default is empty?

```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ Features:
 * server: New `extraPorts` option for adding ports to the Vault server statefulset [GH-841](https://github.com/hashicorp/vault-helm/pull/841)
 * server: Add configurable Port Number in readinessProbe and livenessProbe for the server-statefulset [GH-831](https://github.com/hashicorp/vault-helm/pull/831)
 * injector: Make livenessProbe and readinessProbe configurable and add configurable startupProbe [GH-852](https://github.com/hashicorp/vault-helm/pull/852)
+* CSI: Make `nodeSelector` and `affinity` configurable for CSI daemonset's pods [GH-862](https://github.com/hashicorp/vault-helm/pull/862)
```
Comment (tomhjp) [inline CHANGELOG.md:13]: Please can you add a new Features heading in the Unreleased section for this?

**Review by tomhjp — COMMENTED** [review]
The changes look really good thanks! I'm happy to merge with a couple of small updates if possible please.

**Review by tomhjp — APPROVED** [review]
Thanks!

**Comment by sgargan** [comment]
Thanks for this, any idea when this might make it into a release to the helm repo?

**Pull request merged**